src/Sylius/Behat/Context/Api/Shop/AccountContext.php

```diff
@@ -0,0 +1,203 @@
+use Sylius\Behat\Service\SharedStorageInterface;
+use Sylius\Component\Core\Model\ShopUserInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Webmozart\Assert\Assert;
+
+final class AccountContext implements Context
+{
+    /** @var ApiClientInterface */
+    private $accountClient;
+
+    /** @var SharedStorageInterface */
+    private $sharedStorage;
+
+    /** @var ResponseCheckerInterface */
+    private $responseChecker;
+
+    /** @var ShopSecurityContext */
+    private $shopApiSecurityContext;
+
+    public function __construct(
+        ApiClientInterface $accountClient,
+        SharedStorageInterface $sharedStorage,
+        ResponseCheckerInterface $responseChecker,
+        ShopSecurityContext $shopApiSecurityContext
+    ) {
+        $this->accountClient = $accountClient;
+        $this->sharedStorage = $sharedStorage;
+        $this->responseChecker = $responseChecker;
+        $this->shopApiSecurityContext = $shopApiSecurityContext;
+    }
+
+    /**
+     * @When I want to modify my profile
+     */
+    public function iWantToModifyMyProfile(): void
+    {
+        /** @var ShopUserInterface $shopUser */
+        $shopUser = $this->sharedStorage->get('user');
+        $customer = $shopUser->getCustomer();
+
+        $this->accountClient->buildUpdateRequest((string) $customer->getId());
+    }
+
+    /**
+     * @When I specify the first name as :firstName
+     * @When I remove the first name
+     */
+    public function iSpecifyTheFirstName(string $firstName = ''): void
+    {
+        $this->accountClient->addRequestData('firstName', $firstName);
+    }
+
+    /**
+     * @When I specify the last name as :lastName
+     * @When I remove the last name
+     */
+    public function iSpecifyTheLastName(string $lastName = ''): void
+    {
+        $this->accountClient->addRequestData('lastName', $lastName);
+    }
+
+    /**
+     * @When I specify the customer email as :email
+     * @When I remove the customer email
+     */
+    public function iSpecifyCustomerTheEmail(string $email = ''): void
+    {
+        $this->accountClient->addRequestData('email', $email);
+    }
+
+    /**
+     * @When I save my changes
+     * @When I try to save my changes
+     */
+    public function iSaveMyChanges(): void
+    {
+        $this->accountClient->update();
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully edited
+     */
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyEdited(): void
+    {
+        Assert::true($this->responseChecker->isUpdateSuccessful($this->accountClient->getLastResponse()));
+    }
+
+    /**
+     * @Then my email should be :email
+     * @Then my email should still be :email
+     */
+    public function myEmailShouldBe(string $email): void
+    {
+        /** @var ShopUserInterface $shopUser */
+        $shopUser = $this->sharedStorage->get('user');
+
+        $this->shopApiSecurityContext->iAmLoggedInAs($email);
```

Comment:
This is different behaviour than in UI, for now, it's ok, but we should probably think about it in future

Reply:
Yes, i will investigate how to save `token` after customer change login, and i will add new PR with solution or add issue for it.